lib/charms/tls_certificates_interface/v2/tls_certificates.py

```diff
@@ -1242,9 +1242,9 @@ def certificate_issued_for_csr(self, app_name: str, csr: str) -> bool:
             bool: True/False depending on whether a certificate has been issued for the given CSR.
         """
         issued_certificates_per_csr = self.get_issued_certificates()[app_name]
-        for request_pair in issued_certificates_per_csr:
-            if "csr" in request_pair:
-                return csr_matches_certificate(csr, request_pair["certificate"])
+        for issued_pair in issued_certificates_per_csr:
+            if "csr" in issued_pair and issued_pair["csr"] == csr:
```

Comment:
Since [you identified that there are no tests for this](https://github.com/canonical/tls-certificates-interface/pull/64#discussion_r1308189910), should we add one now?

Reply:
Done